vertx-web/src/main/java/io/vertx/ext/web/impl/HttpServerRequestWrapper.java

```diff
@@ -19,14 +23,50 @@
   private String path;
   private String uri;
   private String absoluteURI;
-
+  private ParsedMIMES parsedHeaders;
+
+  static class ParsedMIMES{
+    public final List<ParsedMIME> accept;
+    public final ParsedMIME contentType;
+    public ParsedMIMES(List<ParsedMIME> accept, ParsedMIME contentType) {
+      this.accept = accept;
+      this.contentType = contentType;
+    }
+  }
+  
   HttpServerRequestWrapper(HttpServerRequest request) {
     delegate = request;
     method = request.method();
     path = request.path();
     uri = request.uri();
     absoluteURI = null;
+    
+    // MYTODO for POC convenience, this is calculated here. In reality, it should be lazy
+    processHeaders(delegate.headers());
+  }
+ 
+  
+  private void processHeaders(MultiMap source) {
+    
+    @Nullable
+    String acceptHeader = source.get("Accept");
+    String contentType = source.get("Content-Type");
+    
+    if(acceptHeader == null){
+      acceptHeader = "*/*";
+    }
+    if(contentType == null){
+      // If no contentType is set, then only the "*" may match it
+      contentType = "\0";
+    }
+    
+    parsedHeaders = new ParsedMIMES(
```

Comment:
For example quality can be applied to the following headers:

* accept
* accept-charset
* accept-encoding
* accept-language

pretty much any `accept-*`, so I'd say that it is better to have a common solution that have it just for 2 headers (if possible).